test_runner: add --test-skip-pattern cli option (#52529)
nodejs/node: doc/api/test.md

## Filtering tests by name

The [`--test-name-pattern`][] and [`--test-skip-pattern`][] command-line

> options provide flexibility in selecting which tests to run and which to

Reviewer: ```suggestion
options allow selecting which tests to run and which to
```
"provides flexibility" seems more an opinion than a fact

skip based on their names.

### Using `--test-name-pattern`

> The `--test-name-pattern` option filters tests based on their names by

Reviewer 2: There is a lot of subtle unrelated doc rewriting happening in these sections. Can you undo them and propose them as a separate docs PR to keep this PR easier to review.

Reviewer: Unresolving this, there is still a lot of unrelated documentation changes

Test name patterns can also be specified using regular expression literals. This
allows regular expression flags to be used. In the previous example, starting

> Node.js with `--test-name-pattern="/test [4-5]/i"` (or `--test-skip-pattern`)

Reviewer 3: Could you add the example?
```suggestion
Node.js with `--test-name-pattern="/test [4-5]/i"` (or `--test-skip-pattern="/test [4-5]/i"`)
```

> Node.js with `--test-[name/skip]-pattern="/test [4-5]/i"` would match `Test 4`

Reviewer 2: I don't know if we should make this particular change because new users are likely to copy and paste that.  I'd leave this as is, or if you really want, use only the skip flag here.

});
```

Using `--test-name-pattern="test 1 some test"` would match only `some test` in `test 1`.

### Using `--test-skip-pattern`

> Similarly, the `--test-skip-pattern` option allows skipping tests based on their

Reviewer 2: ```suggestion
The `--test-skip-pattern` option allows skipping tests based on their
```

names by matching them against the provided pattern, interpreted as JavaScript

> regular expressions. If a test's name matches the skip pattern, it will be
> excluded from execution.

Reviewer 2: ```suggestion
regular expressions. If a test's name matches the skip pattern, the test will not be
executed.
```
